P4-grpc-cve-process.md

```diff
@@ -0,0 +1,88 @@
+* Implemented in: N/A
+* Last updated: 2019-04-09
+* Discussion at: https://groups.google.com/d/msg/grpc-io/jhdCyvKfZlk/HUV675QwBQAJ
+## Abstract
+
+This gRFC proposes the CVE (Common Vulnerabilities and Exposure) process for gRPC. It is heavily adopted and influenced by similar processes that are in place for other Cloud Native Computing Foundation (CNCF) projects such as Kubernetes and Envoy.
+
+
+## Proposal
+
+This proposal defines the Common Vulnerabilities and Exposure (CVE) process for gRPC.  It is heavily adopted and influenced by similar processes that are in place for Kubernetes and Envoy.
+
+## Why do we need a CVE Process?
+
+gRPC is an RPC library with a 6 week release cycle that is in use by major companies, products and services today. The impact of a vulnerability can therefore have large economic, privacy and security implications.  
+
+A CVE disclosure process allows people to responsibly disclose these issues, and for the gRPC team to proactively address the issues and manage the communication around the disclosure. 
+
+
+## Disclosures
+
+### Private Disclosure Process
+
+The gRPC Community asks that all suspected vulnerabilities be privately and responsibly disclosed via the Private Disclosure process described below:
+
+To make a report, email the private grpc-security@googlegroups.com list with the security details and the [details expected for all gRPC bug reports](https://github.com/grpc/grpc/blob/master/.github/ISSUE_TEMPLATE.md).
+
+#### When should I report a vulnerability
+* You think you have discovered a potential security vulnerability in gRPC
+* You are unsure how a vulnerability affects gRPC
+* You think you discovered a vulnerability in another project that gRPC depends on.
+
+#### When should I not report a vulnerability
+* You need help tuning gRPC security
+* You need help applying security related updates
+* Your issue is not security related
+
+#### Security Vulnerability Response
+Each report is acknowledged and analyzed by Product Security Team (PST) members within 3 working days. 
+
+Any vulnerability information shared with PST stays within the gRPC project and will not be disseminated to other projects unless it is necessary to get the issue fixed.
+
+As the security issue moves from triage, to identified fix, to release planning we will keep the reporter updated.
+
+
+#### Public Disclosure Timing
+A public disclosure date is negotiated by the gRPC PST and the bug submitter. We prefer to fully disclose the bug as soon as possible once a user mitigation is available. It is reasonable to delay disclosure when the bug or the fix is not yet fully understood, the solution is not well-tested, or for vendor coordination. The timeframe for disclosure is from immediate (especially if it’s already publicly known) to a few weeks. As a basic default, we expect report date to disclosure date to be on the order of 7 days. The gRPC product security team holds the final say when setting a disclosure date.
```

Comment:
Can you describe how vendors can be added to the gRPC notification list?  Do some projects qualify for advanced notification? Envoy, for example, depends on gRPC and affects many downstream projects in turn. Will you provide advanced notification of fixes to allow us to prepare, or should we follow grpc-io@googlegroups.com? Could you create a list just for security announcements that we can track?

CC @envoyproxy/maintainers

Reply:
at this point, we had no plans to provide advanced notification. We wanted to roll out this process and tweak it as necessary to provide such notification. 

We debated a separate security list, but opted to publish broadly to a single list, so grpc-io@ would be the right group to subscribe to. For security issue, we would likely create an attention grabbing subject to make sure that users view it.

Reply:
Can you specify at least a standard subject line format? I know personally I would not read every in grpc-io@, but I would write a filter that would bring to my attention the security posts and sinkhole the rest of the list for this purpose.

Reply:
A special subject line sounds good. Given the very low number of security issues we have seen in the past having a separate group at this time is probably not needed.

Reply:
Specified this subject line

Reply:
Would it make sense to send security posts to grpc-io+security@googlegroups.com.  Everyone subscribed to grpc-io@ would still get the email but it would be easier to filter security posts based on address.

Reply:
Both are fine as it requires the user to setup a filter anyway. I think having [Security...] in the subject line has a higher chance of getting attention than having +security in To address. Also, I am not sure how prevalent is the use of "+" in email clients other than gmail. I know it is not a common in Outlook.